samples/boards/esp32/flash_memory_mapped/src/main.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/printk.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/flash.h>
+#include <zephyr/storage/flash_map.h>
+
+#include <esp_spi_flash.h>
+#include <soc.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(flash_memory_mapped, CONFIG_LOG_DEFAULT_LEVEL);
+
+int main(void)
+{
+	uint8_t buffer[32];
+	const struct device *flash_device;
+	const void *mem_ptr;
+	spi_flash_mmap_handle_t handle;
+	off_t address = FIXED_PARTITION_OFFSET(scratch_partition);
+	size_t size = FIXED_PARTITION_SIZE(scratch_partition);
+
+	flash_device = DEVICE_DT_GET(DT_CHOSEN(zephyr_flash_controller));
+	if (!device_is_ready(flash_device)) {
+		printk("%s: device not ready.\n", flash_device->name);
+		return 0;
+	}
+
+	/* map selected region */
+	spi_flash_mmap(address, size, SPI_FLASH_MMAP_DATA, &mem_ptr, &handle);
```

Comment:
here it would be useful to check the proper (64kB) alignment of the source address. Something like `if (((64*1024)-1) & address)`